src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySourceConfig.java

```diff
@@ -171,6 +180,26 @@ public String getPartitionTo() {
     return Strings.isNullOrEmpty(partitionTo) ? null : partitionTo;
   }
 
+  @Nullable
+  public String getFilter() {
+    if (filter != null) {
+      filter = filter.trim();
+      if (filter.isEmpty()) {
+        return null;
+      }
+      // remove the WHERE from the filter
+      if (filter.toUpperCase().startsWith(WHERE)) {
+        filter = filter.replace(WHERE, "");
+      }
+    }
+    return filter;
+  }
+
+  @Nullable
+  public void setFilter(final String filter) {
```

Comment:
config classes should basically never have setters. They represent the config properties provided by the user, and should not be modifiable.

This also looks like it's not used.